src/jdk.jlink/share/classes/jdk/tools/jlink/internal/Platform.java

```diff
@@ -50,8 +50,9 @@ public static Platform parsePlatform(String platformString) {
         OperatingSystem os = OperatingSystem.valueOf(osName.toUpperCase(Locale.ROOT));
 
         archName = platformString.substring(index + 1);
-        // Alias architecture "amd64" to "X64"
-        archName = archName.replace("amd64", "X64");
+        // Alias architecture "amd64" to "X64" and "ppc64le" to "ppc64"
+        archName = archName.replace("amd64", "X64")
+                           .replace("ppc64le", "ppc64");
```

Comment:
Is linux-ppc64 big endian? If so, a module with a ModuleTarget value of linux-ppc64le can't be linked with a module that has ModuleTarget value of linux-ppc64.  I'm not say the aliasing here is wrong but I am curious if attempts to link will fail.

Reply:
Yes, linux-ppc64 is big endian. The 2 flavors are completely incompatible. linux-ppc64le is basically a new platform.

Reply:
> Yes, linux-ppc64 is big endian. The 2 flavors are completely incompatible. linux-ppc64le is basically a new platform. Endianness should actually get checked in addition to the archName.

Okay, I was only asking because the ModuleTarget attribute has the OS and arch, the endianness is derived from those right now. Maybe in the future there can be more about the architecture and processor features (e.g. soft vs. hard float).  So mulling over this change and wondering what happens if there is cross linking involve these two platforms.

Reply:
Hmm,  most java bytecode (classfiles) aren't going to care about the endian-ness; in int is an int, etc.
Runtime calls to extract int from byte[] is going to check a runtime flag; possibly static and optimization by HotSpot, etc.
The Architecture enum was intending to capture essential build time attributes.
If there is a need for more static attributes of a build, the current API doesn't cover that.

Reply:
> Hmm, most java bytecode (classfiles) aren't going to care about the endian-ness; in int is an int, etc.
> Runtime calls to extract int from byte[] is going to check a runtime flag; possibly static and optimization by HotSpot, etc.
> The Architecture enum was intending to capture essential build time attributes.
> If there is a need for more static attributes of a build, the current API doesn't cover that.

The image writer needs to be told the endianness of the target platform, usually the native order but can be overridden. As part of pull/11943, a new resource file is added to map the value of the ModuleTarget class file attribute (essentially $os-$arch) to the endianness to help cross linking.  I don't yet see how this will work with the changes in the this PR.

Reply:
Thanks for the hint! I've added a comment to https://github.com/openjdk/jdk/pull/11943 .